botbuilder-instumentation/Instumentation/BotFrameworkInstrumentation.cs

```diff
@@ -131,6 +132,19 @@ public void TrackQnaEvent(IActivity activity, string userQuery, string kbQuestio
             }
         }
 
+        public void TrackGoalTriggeredEvent(IActivity activity, string goalName,
+            IDictionary<string, string> goalTriggeredEventProperties = null, string eventName = TelemetryEventTypes.GoalTriggeredEvent)
+        {
+            if(goalTriggeredEventProperties == null)
+                goalTriggeredEventProperties = new ConcurrentDictionary<string, string>();
```

Comment:
@garypretty , `ConcurrentDictionary` is thread safe but has very poor performance, and since this code is thread safe and the dictionary is created and destroyed inside this method scope, I am not sure why to use it instead of the regular dictionary.  I can't see how this resource is vulnerable by being shared by other threads.